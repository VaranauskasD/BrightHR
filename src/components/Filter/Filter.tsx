import React from 'react'
import styled from 'styled-components'

import { FiFilter } from 'react-icons/Fi'

interface FilterProps {}

const SearchWrapper = styled.div`
  display: flex;
  align-items: center;
  background: ${(props) => `${props.theme.colors.secondary}`};
  border-radius: 24px;

  :focus-within {
    outline: 3px solid black;
  }
`

const StyledLabel = styled.label`
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
  overflow: hidden;
  clip: rect(0 0 0 0);
`

const StyledFilterIcon = styled(FiFilter)`
  margin: 0 4px 0 8px;
`

const StyledInput = styled.input`
  border-radius: 0 24px 24px 0;
  border: none;
  min-width: 330px;
  height: 100%;
  margin-left: 4px;
  padding-left: 4px;
  background: ${(props) => `${props.theme.colors.secondary}`};
  outline: none;
  border-left: 1px solid ${(props) => `${props.theme.colors.grey}`};
`

export const Filter = (props: FilterProps) => {
  return (
    <SearchWrapper>
      <StyledLabel>Filter files</StyledLabel>
      <StyledFilterIcon />
      <StyledInput type="text" placeholder="Filter"></StyledInput>
    </SearchWrapper>
  )
}

export default Filter
