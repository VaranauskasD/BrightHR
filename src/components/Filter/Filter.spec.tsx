import { customRender as render } from '../../utils/test'
import { Filter } from './Filter'

describe('Filter', () => {
  it('renders correctly', async () => {
    const { asFragment } = render(<Filter />)
    expect(asFragment()).toMatchSnapshot()
  })
})
