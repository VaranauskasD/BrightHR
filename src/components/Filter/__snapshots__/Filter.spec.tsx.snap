// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ hbsOQq"
  >
    <label
      class="sc-gsnTZi fDnEkf"
    >
      Filter files
    </label>
    <svg
      class="sc-dkzDqf dMgMsg"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="22 3 2 3 10 12.46 10 19 14 21 14 12.46 22 3"
      />
    </svg>
    <input
      class="sc-hKMtZM lchXNC"
      placeholder="Filter"
      type="text"
    />
  </div>
</DocumentFragment>
`;
